### PR TITLE
Make sure categorical data stays categorical in par synthesizer

### LINKED ISSUE #1910: PAR DiagnosticReport not 1.0 with float categorical columns
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version:
* Python version:
* Operating System:

### Error Description

When running PAR with categorical columns that are floats, PAR does not stick to the original categories when sampling. This leads to a very low diagnostic score for `'Data Validity'` due to the `CategoryAdherence` metric failing.

### Steps to reproduce
```python
from sdv.datasets.demo import download_demo
from sdv.sequential import PARSynthesizer
from sdv.evaluation.single_table import run_diagnostic

data, metadata = download_demo('sequential', 'nasdaq100_2019')
data['category'] = [100.0 if i % 2 == 0 else 50.0 for i in data.index]
metadata.add_column('category', sdtype='categorical')

synth = PARSynthesizer(metadata)
synth.fit(data)
sampled = synth.sample(2)

report = run_diagnostic(data, sampled, metadata)
```

### PR DESCRIPTION
**Pull request by lajohn4747**
resolve #1910 
CU-86b01mjx8

Ensures that categorical labeled columns do not get altered by the inferred type by pandas.

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b01mjx8 SDV - PAR DiagnosticReport not 1.0 with float categorical columns #1910](https://app.clickup.com/t/86b01mjx8)

File: sdv/sequential/par.py [hunk 1]
```diff
@@ -380,6 +380,10 @@ def _fit_sequence_columns(self, timeseries_data):
             kind = dtype.kind
             if kind in ('i', 'f'):
                 data_type = 'continuous'
+                # Check if metadata overrides this data type
+                if field in self.metadata.columns:
+                    if self.metadata.columns[field].get('sdtype', None) == 'categorical':
```
Comment (amontanez24) [inline sdv/sequential/par.py:385]:
```suggestion
                if self.metadata.columns.get(field, {}).get('sdtype', None) == 'categorical':
```

File: sdv/sequential/par.py [hunk 2]
```diff
@@ -380,6 +380,10 @@ def _fit_sequence_columns(self, timeseries_data):
             kind = dtype.kind
             if kind in ('i', 'f'):
                 data_type = 'continuous'
+                # Check if metadata overrides this data type
```
Comment (amontanez24) [inline sdv/sequential/par.py:383]: we can add a unit test that mocks `self._model` and makes sure it's called with categorical in the right index of data_types for this case

**Review by amontanez24 — COMMENTED** [review]
I think this works! @frances-h not sure if you have any concerns with this approach

**Review by frances-h — APPROVED** [review]
Looks good!

**Review by amontanez24 — APPROVED** [review]

**Pull request merged**